themes/src/main/resources/theme/base/login/resources/js/webauthnAuthenticate.js

```diff
@@ -0,0 +1,78 @@
+import { base64url } from "../../../common/keycloak/node_modules/rfc4648/lib/rfc4648.js";
+
+export function authenticateByWebAuthn(input) {
+    if (!input.isUserIdentified) {
+        doAuthenticate([], input.challenge, input.userVerification, input.rpId, input.createTimeout, input.errmsg)
+            .then(result => returnSuccess(result))
+            .catch(err => returnFailure(err));
+        return;
+    }
+    checkAllowCredentials(input.challenge, input.userVerification, input.rpId, input.createTimeout, input.errmsg);
+}
```

Comment:
Let's invert this condition to create an early return and avoid nesting, as well as using `async`/`await` to handle the Promise.


```suggestion
export async function authenticateByWebAuthn(input) {
    if (input.isUserIdentified) {
        checkAllowCredentials(input.challenge, input.userVerification, input.rpId, input.createTimeout, input.errmsg);
        return;
    }

    try {
        const result = await doAuthenticate([], input.challenge, input.userVerification, input.rpId, input.createTimeout, input.errmsg);
        returnSuccess(result)
    } catch (error) {
        returnFailure();
    }
}
```

Reply:
I fixed the codes as you suggested.